### DISCUSSION TITLE
Trying to subset HRRR Dataset with xarray

### DISCUSSION THREAD

**Discussion by v7nguyen** [blaylockbk/Herbie#92] [Q&A]
Hi Brian,

Recently, I have run into some issues with my system not being able to allocate enough memory in order to run ds.to_dataframe(). 

As a result, I am trying to subset the xarray dataset by min/max lat/lons before converting it to a dataframe. At this point, I've exhausted everything I find to try:

![image](https://user-images.githubusercontent.com/53784737/183474034-7bdacf39-7209-483b-a161-741835a0c48b.png)

![image](https://user-images.githubusercontent.com/53784737/183474110-465b8813-69bb-4840-848f-dcb7ccf52e16.png)

![image](https://user-images.githubusercontent.com/53784737/183479753-1a39bc81-c688-44b3-a7d3-c8a5e66359f8.png)
![image](https://user-images.githubusercontent.com/53784737/183479805-3bc16c6c-ec8b-4979-9174-d3b7699149e1.png)
![image](https://user-images.githubusercontent.com/53784737/183479867-5234f370-cda3-4d99-97db-9b8d9217c531.png)
![image](https://user-images.githubusercontent.com/53784737/183479954-db9218a7-6cd4-41fc-937b-c56d23afd976.png)

Can you provide some insight how to achieve this? I hypothesize that xarray is having issues with the Lambert Conformal projection format the lat/lons are in. I am confused why the ds.where() function is failing on these coordinates. Would you be able to try and reproduce this?

**Comment by kylejgillett** [comment]
Hey @blaylockbk, I too am running into a similar issue here with attempting to subset a lat/lon box post-download. Any thoughts on this issue?

**Comment by joshuaeh** [comment]
I had a similar issue some time ago and sometimes found it helpful to shift my longitude range (-180, 180) <-> (0, 360). This was just a band-aid but I never got around to diagnosing the issue.

**Comment by williamhobbs** [comment]
Here's a trial-and-error approach I have used to select subsets of the HRRR. I just needed to pick a reasonable area once, so it was good enough for my use, even if it isn't "optimal."

```python
ds = Herbie(date='2021-06-19 06:00', model='HRRR', product="sfc", fxx=36).xarray('DSWRF',remove_grib=True)
```
![image](https://github.com/blaylockbk/Herbie/assets/45701090/512c28cf-d00e-4dad-b4e5-87a48cd8f886)


That returned a preview of the xarray dataset. 

We will want to slice the datasets (subset them) to reduce file size. Ideally, we would do this based on latitude and longitude ranges, but the HRRR does not use Cartesian coordinates (see https://github.com/blaylockbk/Herbie/discussions/45). So, we will use trial and error to find x and y ranges that cover our region of interest (SE US, in my case). Note that y ranges from 0-1059, and x ranges from 0-1799 (see dimensions in the output above).

Modify the ranges used in `x=slice(xmin, xmax), y=slice(ymin, ymax)` below.

```python
xmin, xmax = 1080, 1440
ymin, ymax = 220, 440
ds_cropped = ds.sel(x=slice(xmin, xmax), y=slice(ymin, ymax))
```
Generate a simple plot:

```python
ax = EasyMap().STATES().OCEAN().LAND().DOMAIN(ds_cropped).ax
ds_cropped.dswrf.plot(
    x="longitude",
    y="latitude",
    ax=ax,
    transform=pc,
    cbar_kwargs={"shrink": 0.5, "orientation": "horizontal", "pad": 0.01},
)
```
![image](https://github.com/blaylockbk/Herbie/assets/45701090/04acbaeb-e25d-43cd-9405-474565fa2177)

Go back up and modify the slice limits (`xmin`, `xmax`, `ymin`, `ymax`) and re-run as needed.
